MetaMask/metamask-extension
# Buy crypto by redirecting to onramp experience on pdapp instead of deposit popover

## Explanation
This pull request changes the implementation and the experience of buying crypto assets on MetaMask extension, which currently relies on hardcoded integration with a set of predefined providers (Coinbase Pay, Transak, MoonPay), to a brand-new implementation that improves the overall user experience by redirecting MetaMask users to the onramp curated experience on the MetaMask Portfolio Dapp (https://portfolio.metamask.io/buy).

With the new approach, we can achieve greater flexibility and better coverage by being able to streamline our operations dynamically and offer integration with new providers and crypto assets without the need for a new deployment, which would otherwise cause delays.

## Changes

- The previous hardcoded integration with predefined providers has been removed
- A new implementation that redirects the user to the MetaMask Portfolio Dapp for a better, curated buying experience has been added when clicking the buy button anywhere in the extension.
- The new implementation follows best practices for security and usability, ensuring a smooth and seamless user experience



## Screenshots/Screencaps



### Before

https://user-images.githubusercontent.com/2320312/220421951-4ec02c68-b057-4065-8763-6beff6f3f709.mov




### After

https://user-images.githubusercontent.com/2320312/220422041-a9e9a15c-2dbd-473a-a3c9-232b28b679fd.mov



## Manual Testing Steps

1. Navigate to the buying page
2. Initiate a buy action
3. Verify that you are redirected to the MetaMask Portfolio Dapp



## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
